src/ui/public/saved_object_table/crud_item_table.js

```diff
@@ -0,0 +1,182 @@
+import React from 'react';
+import _ from 'lodash';
+
+import { ItemSelectionActions } from 'ui/saved_object_table/item_selection_actions';
+import { sortItems, getFlippedSortOrder } from 'ui/saved_object_table/sort_items';
+import { Pager, ToolBarPager } from 'ui/pager';
+import { ListingTableActions } from 'ui/saved_object_table/listing_table_actions';
+import { getCheckBoxColumn } from './get_checkbox_column';
+
+import {
+  ItemTable,
+  DeleteButton,
+  CreateButtonLink,
+  SortOrder,
+  KuiToolBarSearchBox,
+  KuiToolBarPager,
+  KuiToolBar,
+  KuiToolBarFooter,
+  KuiToolBarText,
+  KuiTable,
+  KuiTh,
+  KuiTHead,
+  KuiTBody,
+  KuiTr,
+  ItemTableRow,
+} from 'ui_framework/components';
+
+
+const ITEMS_PER_PAGE = 2;
+
+export class CrudItemTable extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = ListingTableActions.getInitialState();
+    this.pager = new Pager(ITEMS_PER_PAGE);
+  }
+
+  componentDidMount() {
+    this.onFilter();
+  }
+
+  onFilter = (newFilter) => {
+    this.setState(ListingTableActions.startFetching());
+    this.props.fetch(newFilter).then((results) => {
+      const items = results.hits;
+      const lastPageIndex = this.pager.getLastPageIndex(items.length);
+      this.setState(ListingTableActions.setFilteredItems(this.state, items, lastPageIndex));
+    }).finally(() => {
+      this.setState(ListingTableActions.endFetching());
+    });
+  };
+
+
+  getPageOfItems = () => this.pager.getItemsOnPage(this.state.items, this.state.currentPageIndex);
+
+  onToggleAll = () => this.setState(ListingTableActions.onToggleAll(this.state, this.getPageOfItems()));
```

Comment:
I tried to document that naming scheme in https://github.com/elastic/kibana/pull/9957/files#diff-c344794f6f396d6c80fd35cdc8eb5bb1R1029. But IMHO in this case it would mean, that all the "action" methods are named using strong, imperative verbs. The `on...` scheme applies to the names of props that are used to communicate some kind of user action to the rest of the system, for example:

```javascript
turnToNextPage() {
  this.setState(...);
}

render() {
  return (
    <Pager onTurnToNextPage={this.turnToNextPage} />
  );
}
```

So `turnToNextPage` would already be correctly named, but `onToggleAll` would change to `toggleAll`.